torch/utils/_python_dispatch.py

```diff
@@ -65,6 +65,9 @@ def enable_torch_dispatch_mode(mode, *, replace=None, ignore_preexisting=False)
 def _wrap_torch_dispatch(f):
     @functools.wraps(f)
     def wrapped(self, *args, **kwargs):
+        if isinstance(f, classmethod):
+            raise RuntimeError("TorchDispatchMode's torch_dispatch function " +
+                               "should be a normal method not a class method")
```

Comment:
This is done that the error shows up on the usage of the function, not the wrapping (which happens at class declaration time). Happy to move it if we want it to happen when the class is declared but this felt more in line with what a user would expect from Python